homeassistant/components/huawei_lte/__init__.py

```diff
@@ -131,16 +138,16 @@ def get_data(self, config):
         return None
 
 
-def setup(hass, config) -> bool:
+async def async_setup(hass, config) -> bool:
```

Comment:
Why do we change to the async api?

Reply:
Because I'll be moving this component towards config entry based setup, and I believe there's no sync option available there. Also I suppose the smaller threadsafe/executor wrapped sections the better in general.

Reply:
Fewer context switches is more efficient so if the library doesn't support asyncio we recommend to stay with the sync api.

`async_setup_entry` is a coroutine, yes. `setup` can still be kept a regular function if needed.